analyzer/analyzer.go

```diff
@@ -141,7 +165,7 @@ func (n *perfSprint) run(pass *analysis.Pass) (interface{}, error) {
 
 		switch verb {
 		default:
-			if fn == "fmt.Sprintf" && isConcatable(verb) && n.strconcat {
+			if fn == "fmt.Sprintf" && isConcatable(verb) && n.strFormat.strconcat {
```

Comment:
I have a problem here, and at some other place

Shouldn't be this?

```suggestion
			if fn == "fmt.Sprintf" && isConcatable(verb) && n.strFormat.strconcat && n.strFormat.global {
```

Because at some other place .global is checked

So, it leads to me raise a point.

- Shouldn't you have methods on the structs like strFormat one check to return a boolean that will check global then strconcat ?
- or should there be a code, at init level check for global being false, and then set "depending" flags to false. Like setting n.strFormat.strconcat=false and n.strFormat.sprint1=false when n.strFormat.global == false

The second option would help to keep code clean and do not have to check the global flag, and avoid adding methods.

Reply:
Nice, fixing it the second way